src-plugins/mscAlgorithmPaint/mscAlgorithmPaintToolBox.cpp

```diff
@@ -608,36 +608,60 @@ void AlgorithmPaintToolBox::updateView()
 
     if (view)
     {
-        setCurrentView(qobject_cast<medAbstractImageView*>(view));
-
-        updateMouseInteraction();
-
-        medAbstractData* data = currentView->layerData(0);
-
-        if(data)
+        medAbstractImageView * v = qobject_cast<medAbstractImageView*>(view);
+        if (v)
         {
-            medImageMaskAnnotationData * existingMaskAnnData = dynamic_cast<medImageMaskAnnotationData *>(data);
-            if(!existingMaskAnnData)
+            // AlgorithmPaintToolBox does not work with meshes
+            for (unsigned int i=0 ; i<v->layersCount() ; ++i)
             {
-                setData( data );
+                medAbstractData *data = v->layerData(i);
+                if(!data || data->identifier().contains("vtkDataMesh"))
+                {
+                    qDebug()<<"AlgorithmPaintToolBox::updateView() do not use meshes in this toolbox.";
+                    setButtonsDisabled(true);
+                    isMeshData=true;
+                    return;
+                }
+                else
```

Comment:
```
else
{
blabla
}
```
And i added that you'll have to change the structure maybe of this function, so, you'll see if an `else` condition is still needed at the end :)